pkg/cmdexecutor/cmdexecutor.go

```diff
@@ -90,9 +90,8 @@ func (c *cmdExecutor) Start(errChan chan error) error {
 			err = fmt.Errorf("failed to run command: %s in pod: [%s] %s due to err: %v",
 				command, c.podNamespace, c.podName, err)
 			logrus.Errorf(err.Error())
+			errChan <- err
 		}
-
-		errChan <- err
```

Comment:
Let's send the `err` to the channel only if it's not `nil`. This is what fixes the nil pointer dereference error.